src/shell/data-staging.cpp

```diff
@@ -23,97 +23,116 @@ extern "C" {
+        auto granularity = storage_attr.at ("granularity").get_ref<
+            const std::string &>();
         if ((granularity != "node") && (granularity != "job")) {
-            shell_log_error ("Storage entry contains a granularity not 'job' or 'node' (%s)", granularity.c_str());
-            return JobspecState::invalid;
+            shell_log_error ("Storage entry contains a granularity not "
+                             "equal to 'job' or 'node' (%s)",
+                             granularity.c_str());
+            return jobspec_state_t::INVALID;
         }
 
-        json stage_in = storage_attr["stage-in"];
-        if (!stage_in.contains("file")) {
+        json stage_in = storage_attr.at ("stage-in");
+        if (!stage_in.contains ("file")) {
             shell_log_error ("Stage-in entry missing file key");
-            return JobspecState::invalid;
-        } else if (stage_in.size() > 1) {
-            shell_log_error ("Stage-in entry contains extra keys. Only 'file' is valid.");
-            return JobspecState::invalid;
+            return jobspec_state_t::INVALID;
+        } else if (stage_in.size () > 1) {
+            shell_log_error ("Stage-in entry contains extra keys. "
+                             "Only 'file' is valid.");
+            return jobspec_state_t::INVALID;
         }
     }
 
-    return JobspecState::valid_staging;
+    return jobspec_state_t::VALID_STAGING;
 }
 
 static int copy_data (const fs::path &source, const fs::path &destination)
 {
     boost::system::error_code ec;
-    fs::copy_file (source, destination, fs::copy_option::overwrite_if_exists, ec);
+    fs::copy_file (source,
+                   destination,
+                   fs::copy_option::overwrite_if_exists,
+                   ec);
     if (ec) {
-        shell_log_error ("Error staging in data. Error code %d: %s", ec.value(), ec.message().c_str());
+        shell_log_error ("Error staging in data. Error code %d: %s",
+                         ec.value (), ec.message ().c_str ());
         return -1;
     }
     shell_debug ("Completed staging in data.");
     return 0;
 }
 
-static boost::optional<fs::path> get_mountpoint (json resource)
+static boost::optional<fs::path> get_mountpoint (json &resource)
 {
     // TODO: go from LinuxDeviceID in resource -> mountpoint in `resource`
     try {
-        return fs::path(resource["properties"]["mountpoint"].get<std::string>());
+        auto key = "/properties/mountpoint"_json_pointer;
+        return fs::path (resource.at (key).get_ref<const std::string&>());
     } catch (json::exception& e) {
         return boost::none;
     }
 }
 
-static std::map<std::string, json> find_storage_resources (json R, int rank)
+static std::map<std::string, json> find_storage_resources (json &R, int rank)
 {
     std::map<std::string, json> storage_resources;
     json jgf;
     try {
-        jgf = R["scheduling"]["graph"];
+        jgf = R.at("/scheduling/graph"_json_pointer);
```

Comment:
Nit: space before parenthesis.
```suggestion
        jgf = R.at ("/scheduling/graph"_json_pointer);
```